### PR TITLE
Feature/refactor self method

### PR DESCRIPTION
**Pull request by MDelarosa1993**
Refactored the self method so it wouldnt handle the bulk of creating object so i created two class methods to help from_csv and i also made helper methods and tested them for more tackles and fewest tackles.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -118,87 +124,69 @@ def average_goals_by_season
...
+    @teams.each do |team|
+        team_id = team.team_id
+        team_name = team.team_name
+        team_names_hash[team_id] = team_name
+  end
+      team_names_hash  
+end
+  
+  
+  def most_tackles(season)
```
Comment (TDManning) [inline lib/stat_tracker.rb:171]: These methods are much more succinct and DRY. Nice job.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -8,22 +8,28 @@ class StatTracker
...
-    end
-    CSV.foreach(locations[:game_teams], headers: true, header_converters: :symbol) do |row|
-      game_teams << GameTeam.new(row.to_h)
-    end
+    # The from_csv method initializes StatTracker object
+    # with the arrays of Game, Team, and GameTeam objects created from the CSV files. 
+    # The read_and_process_csv method abstracts the common logic of reading and creating objects from a CSV file.
+    # Calls read_and_process_csv for each type of CSV file 
+    # (games, teams, game_teams), passing the appropriate class (Game, Team, GameTeam) to construct the objects.
+    games = read_and_process_csv(locations[:games], Game)
```
Comment (TDManning) [inline lib/stat_tracker.rb:16]: The read_and_process_csv is a creative way to approach this (reading and shoveling the data). It's doing a lot!

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -8,22 +8,28 @@ class StatTracker
...
+    # Converts each row to a hash (row.to_h), 
+    # initializes an object of class klass with this hash, 
+    # and returns an array of these objects.
+  def self.read_and_process_csv(file_path, file_class )
+    CSV.foreach(file_path, headers: true, header_converters: :symbol).map do |row|
+      file_class.new(row.to_h)
+    end
+  end
+  # self is being used here to seperate the class methods from an instance method.
+  #
```
Comment (reneemes) [inline lib/stat_tracker.rb:32]: Thank you for leaving these comments in here to explain what is going on. Also good idea breaking the method up into two.

**Review by reneemes — APPROVED** [review]

**Comment by jdavalos98** [comment]
Great job on this refactor.  Its a great start on making our code look as DRY as possible. Great use of a helper methods in our from csv. Looks so much cleaner!

**Pull request merged**